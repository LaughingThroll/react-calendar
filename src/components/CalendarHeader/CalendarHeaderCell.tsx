import React from 'react'
import classnames from 'classnames'

import { isWeekend, getDayInBinaryString } from '../../utils/date/index'

interface ICalendarHeaderCell {
  date: Date
}

const CalendarHeaderCell: React.FC<ICalendarHeaderCell> = ({ date }) => {
  return (
    <th
      className={classnames({
        'calendar-header__cell': true,
        'cell-gray': isWeekend(date),
      })}
    >
      <span className="calendar-header__day">{getDayInBinaryString(date)}</span>
      <span className="calendar-header__number">{date.getDate()}</span>
    </th>
  )
}

export default React.memo(CalendarHeaderCell)
