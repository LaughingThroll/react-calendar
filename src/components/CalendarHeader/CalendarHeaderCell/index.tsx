import React from 'react'

import Cell from '../../common/Cells/Cell'

import { getDayInBinaryString } from '../../../utils/date'

import { CellVariants } from '../../../types/cell'

export interface CalendarHeaderCellProps {
  cellDate: Date
  classNames?: string[]
}

const CalendarHeaderCell: React.FC<CalendarHeaderCellProps> = ({ cellDate, classNames = [] }) => {
  return (
    <Cell cellDate={cellDate} variant={CellVariants.TH} classNames={classNames}>
      <span className="calendar-header__day">{getDayInBinaryString(cellDate)}</span>
      <span className="calendar-header__number">{cellDate.getDate()}</span>
    </Cell>
  )
}

export default React.memo(CalendarHeaderCell)
