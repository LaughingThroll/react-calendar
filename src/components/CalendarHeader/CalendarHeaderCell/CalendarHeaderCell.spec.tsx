import { mount } from 'enzyme'
import CalendarHeaderCell, { CalendarHeaderCellProps } from './'

const setUp = (props: CalendarHeaderCellProps) => mount(<CalendarHeaderCell {...props} />)

describe('CalendarHeaderCell render', () => {
  it('default render', () => {
    const props = { cellDate: new Date('2021-05-04') }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()

    expect(tree.text().includes('Tu')).toBe(true)
    expect(tree.text().includes('4')).toBe(true)
  })
})
