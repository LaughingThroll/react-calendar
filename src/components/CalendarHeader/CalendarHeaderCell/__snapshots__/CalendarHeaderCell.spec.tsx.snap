// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeaderCell render default render 1`] = `
<Memo(CalendarHeaderCell)
  cellDate={2021-05-04T00:00:00.000Z}
>
  <Memo(Cell)
    cellDate={2021-05-04T00:00:00.000Z}
    classNames={Array []}
    variant="th"
  >
    <th
      className="  "
    >
      <span
        className="calendar-header__day"
      >
        Tu
      </span>
      <span
        className="calendar-header__number"
      >
        4
      </span>
    </th>
  </Memo(Cell)>
</Memo(CalendarHeaderCell)>
`;
